### PR TITLE
modified getSharedbyMe function

### PR DESCRIPTION
**Pull request by S-Panta**
This PR add step to return sharelink object created by the user.
Related Issue:https://github.com/owncloud/ocis-php-sdk/issues/113

### CODE REVIEW

File: tests/integration/Owncloud/OcisPhpSdk/ResourceInviteTest.php [hunk 1]
```diff
@@ -255,14 +256,17 @@ public function testGetReceiversOfShareReturnedBySharedByMe(): void
...
         for($i = 0; $i < 6; $i++) {
-            $this->assertThat(
-                $shares[$i]->getReceiver()->getDisplayName(),
-                $this->logicalOr(
-                    $this->equalTo("philosophy-haters"),
-                    $this->equalTo("Marie Curie"),
-                    $this->equalTo("Albert Einstein")
-                )
-            );
+            if($shares[$i] instanceof ShareCreated) {
```
Comment (individual-it) [inline tests/integration/Owncloud/OcisPhpSdk/ResourceInviteTest.php:259]: there should be only `ShareCreated` items here at this time, or?
The danger with this is, that if we would have a bug and return something else than `ShareCreated` there would be no assertions done.
To make the analyzer happy maybe it would be enough to check `assertContainsOnly`

**Review by individual-it — CHANGES_REQUESTED** [review]
a simple test where there is one link-share and a normal share would be nice

File: tests/integration/Owncloud/OcisPhpSdk/ResourceInviteTest.php (hunk 1 above)
Comment (phil-davis) [reply tests/integration/Owncloud/OcisPhpSdk/ResourceInviteTest.php:259]: There is https://docs.phpunit.de/en/9.6/assertions.html#assertinstanceof

So you could not put this `if` statement, but instead just have an `assertInstanceOf` before `assertThat` - that way the test will fail early if `$shares[$i]` is not an instance of `ShareCreated`

File: tests/integration/Owncloud/OcisPhpSdk/OcisTest.php [hunk 2]
```diff
@@ -535,4 +557,115 @@ public function testGetResourceInvalidId(): void
...
+        foreach ($resources as $resource) {
+            if ($resource->getName() === 'newFolder') {
+                $sharedResource = $resource;
+            }
+        }
+        if(empty($sharedResource)) {
+            throw new \Error(
+                "resource not found "
+            );
+        }
```
Comment (individual-it) [inline tests/integration/Owncloud/OcisPhpSdk/OcisTest.php:579]: use assertion

File: tests/integration/Owncloud/OcisPhpSdk/OcisTest.php [hunk 3]
```diff
@@ -535,4 +557,115 @@ public function testGetResourceInvalidId(): void
...
+                $editorRole = $role;
+            }
+        }
+        if(empty($editorRole)) {
+            throw new \Error(
+                "Editor role not found "
+            );
+        }
+        $sharedResource->invite([$einstein], $editorRole);
+        $myShare = $ocis->getSharedByMe();
```
Comment (individual-it) [inline tests/integration/Owncloud/OcisPhpSdk/OcisTest.php:592]:
```suggestion
        $myShares = $ocis->getSharedByMe();
```

File: tests/integration/Owncloud/OcisPhpSdk/OcisTest.php [hunk 4]
```diff
@@ -535,4 +557,115 @@ public function testGetResourceInvalidId(): void
...
+        $this->assertInstanceOf(ShareCreated::class, $myShare[0]);
+        $this->assertEquals($myShare[0]->getReceiver()->getDisplayName(), 'Albert Einstein');
```
Comment (individual-it) [inline tests/integration/Owncloud/OcisPhpSdk/OcisTest.php:594]: maybe check also the resource name

File: tests/integration/Owncloud/OcisPhpSdk/OcisTest.php [hunk 5]
```diff
@@ -535,4 +557,115 @@ public function testGetResourceInvalidId(): void
...
+                $editorRole = $role;
+            }
+        }
+        if(empty($editorRole)) {
+            throw new \Error(
+                "Editor role not found "
+            );
+        }
+        $sharedResource->invite([$einstein], $editorRole);
+        $sharedResource->createSharingLink(SharingLinkType::VIEW, new \DateTimeImmutable('2023-12-31 01:02:03.456789'), 'Test@123', '');
```
Comment (individual-it) [inline tests/integration/Owncloud/OcisPhpSdk/OcisTest.php:628]: long line

File: tests/integration/Owncloud/OcisPhpSdk/OcisTest.php [hunk 6]
```diff
@@ -535,4 +557,115 @@ public function testGetResourceInvalidId(): void
...
+        $myShare = $ocis->getSharedByMe();
+        $this->assertInstanceOf(ShareCreated::class, $myShare[0]);
+        $this->assertInstanceOf(ShareLink::class, $myShare[1]);
```
Comment (individual-it) [inline tests/integration/Owncloud/OcisPhpSdk/OcisTest.php:631]: Is the order guaranteed ?

File: tests/integration/Owncloud/OcisPhpSdk/OcisTest.php [hunk 7]
```diff
@@ -535,4 +557,115 @@ public function testGetResourceInvalidId(): void
...
+                "Editor role not found "
+            );
+        }
+        $sharedResource->invite([$einstein], $editorRole);
+        $myShare = $ocis->getSharedByMe();
+        $this->assertInstanceOf(ShareCreated::class, $myShare[0]);
+        $this->assertEquals($myShare[0]->getReceiver()->getDisplayName(), 'Albert Einstein');
+    }
+
+    public function testGetShareCreatedAndShareCreatedLinkByMe(): void
```
Comment (individual-it) [inline tests/integration/Owncloud/OcisPhpSdk/OcisTest.php:597]:
```suggestion
    public function testGetSharedByMeContainingShareCreatedAndShareLink(): void
```

File: tests/integration/Owncloud/OcisPhpSdk/OcisTest.php [hunk 8]
```diff
@@ -535,4 +557,115 @@ public function testGetResourceInvalidId(): void
...
+            );
+        }
+        $sharedResource->invite([$einstein], $editorRole);
+        $sharedResource->createSharingLink(SharingLinkType::VIEW, new \DateTimeImmutable('2023-12-31 01:02:03.456789'), 'Test@123', '');
+        $myShare = $ocis->getSharedByMe();
+        $this->assertInstanceOf(ShareCreated::class, $myShare[0]);
+        $this->assertInstanceOf(ShareLink::class, $myShare[1]);
+    }
+
+    public function testGetSharedWithMe(): void
```
Comment (individual-it) [inline tests/integration/Owncloud/OcisPhpSdk/OcisTest.php:634]: How is that related to `getSharedByMe`? Should that be in a different PR?

File: tests/integration/Owncloud/OcisPhpSdk/OcisTest.php [hunk 9]
```diff
@@ -535,4 +557,115 @@ public function testGetResourceInvalidId(): void
         $ocis = $this->getOcis('admin', 'admin');
         $ocis->getResourceById('not-existing-id');
     }
+
+    public function testGetShareCreatedByMe(): void
```
Comment (individual-it) [inline tests/integration/Owncloud/OcisPhpSdk/OcisTest.php:561]:
```suggestion
    public function testGetSharedByMe(): void
```

File: tests/integration/Owncloud/OcisPhpSdk/OcisTest.php [hunk 10]
```diff
@@ -452,6 +457,23 @@ public function testDeleteGroupById(): void
         $this->createdGroups = [$physicsLoversGroup];
     }
 
+    /**
+     * @return void
+     */
+    public function testDeleteGroupByIdNoPermission(): void
```
Comment (individual-it) [inline tests/integration/Owncloud/OcisPhpSdk/OcisTest.php:463]: how is that related to `getSharedByMe`? Should that be a separate PR?

File: tests/integration/Owncloud/OcisPhpSdk/OcisTest.php [hunk 11]
```diff
@@ -535,4 +557,115 @@ public function testGetResourceInvalidId(): void
...
+        foreach($sharedResource->getRoles() as $role) {
+            if($role->getDisplayName() === 'Editor') {
+                $editorRole = $role;
+            }
+        }
+        if(empty($editorRole)) {
+            throw new \Error(
+                "Editor role not found "
+            );
+        }
```
Comment (individual-it) [inline tests/integration/Owncloud/OcisPhpSdk/OcisTest.php:626]: could be moved in `setUp()`

**Review by individual-it — CHANGES_REQUESTED** [review]
please consider to move the tests into an other file `GetSharedByMeTest.php`

File: tests/integration/Owncloud/OcisPhpSdk/OcisTest.php (hunk 6 above)
Comment (S-Panta) [reply tests/integration/Owncloud/OcisPhpSdk/OcisTest.php]: I've tested and found that the order remains `ShareCreated` and `ShareLink`.

**Review by individual-it — APPROVED** [review]
